manager/state/raft/raft.go

```diff
@@ -764,6 +775,28 @@ func (n *Node) CanRemoveMember(id uint64) bool {
 	return n.cluster.CanRemoveMember(n.Config.ID, id)
 }
 
+func (n *Node) transferLeadership(ctx context.Context) error {
+	transferee, err := n.cluster.LongestActive()
+	if err != nil {
+		return errors.Wrap(err, "get longest active member")
+	}
+	start := time.Now()
+	logrus.Infof("raft: transfer leadership %x -> %x", n.Config.ID, transferee)
+	n.TransferLeadership(ctx, n.Config.ID, transferee)
+	ticker := time.NewTicker(100 * time.Millisecond)
+	defer ticker.Stop()
+	for n.leader() != transferee {
```

Comment:
Similar problem here. The leadership transfer won't be triggered until the message it sent out from a `Ready` in the `Run` loop, but the `Run` loop is blocked by this loop.